packages/react-instantsearch/src/core/indexUtils.js

```diff
@@ -125,13 +133,25 @@ export function getCurrentRefinementValue(
   refinementsCallback
 ) {
   const index = getIndex(context);
+  const namespace = getNamespaceAndId(id);
   const refinements =
     (hasMultipleIndex(context) && has(searchState, `indices.${index}.${id}`)) ||
+    (!hasMultipleIndex(context) &&
+      namespace.namespace &&
+      namespace.id &&
+      has(searchState[namespace.namespace], namespace.id)) ||
     (!hasMultipleIndex(context) && has(searchState, id));
```

Comment:
You know the library much better than I do. I left it there just in case I'm missing some use case. I wanted to be 100% sure it's backwards compatible.